static/js/rules/PropertySelect.js

```diff
@@ -156,25 +181,47 @@ PropertySelect.prototype.addOption = function(name, ruleFragment, selected) {
       elt.dataset.ruleFragment = JSON.stringify(ruleFragment);
     });
   } else if (property.type === 'string') {
-    const valueInput = document.createElement('input');
-    valueInput.classList.add('value-input');
-    if (property.name === 'color') {
-      valueInput.type = 'color';
+    let valueInput, valueSelect;
+    if (property.hasOwnProperty('enum') && property.enum.length > 0) {
+      valueSelect = document.createElement('select');
+      valueSelect.classList.add('value-select');
+
+      for (const choice of property.enum) {
+        const option = document.createElement('option');
+        option.value = choice;
+        option.innerText = choice;
+        valueSelect.appendChild(option);
+      }
+
```

Comment:
valueSelect needs a stopPropagation somewhere so that clicking to select a value doesn't open the rule select menu

Reply:
Done.